### ISSUE TITLE
[ASM] a16w16 split-K: the entry-zero counter is unsafe under CUDA graph capture; opus-style fp32 partials would remove the class

### ISSUE THREAD

**Issue by JohnQinAMD** [ROCm/aiter#4920]
## Summary

The a16w16 ASM split-K kernels gate their reduction on an atomic counter that **must be zero when a launch starts**. Blocks count in; the last one reduces and writes the counter back to zero.

A CUDA graph records kernel launches, not the `memset` that zeroed the counter. So on replay nothing re-establishes the entry condition: a counter left non-zero stays non-zero, the reduction never fires, and every wave spins. The queue wedges — no fault, no wrong number, no recovery.

`rocgdb` on a hung endpoint: all 132 waves of `bf16gemm_fp32bf16_tn_64x64_splitk_clean` waiting on a counter holding `2`.

This is a property of the kernel's protocol, not of any one caller. Any host that records a split-K a16w16 launch into a graph hits it.

## What a kernel-side fix would remove

#4916 fixes the hang on the shipped binaries and is what unblocks graphed serving now: each recorded launch gets its own counter from a pool allocated before capture, with the zero-fill recorded in the graph. Throughput is unchanged.

What it cannot do is make the counter unnecessary. Because the counter exists, the host has to size a pool by a constant, raise when it runs out, and raise on a device that never ran eagerly. Removing the counter deletes all three, and the host-side code with them.

## Proposed fix

Drop the counter. Each split writes its fp32 partial to a disjoint slice of a `[split_k, M, N]` workspace, and a separate reduce kernel sums along the split axis, with ordering coming from the stream. No entry precondition, nothing for a graph to restore, nothing for the host to manage.

**This is not a new design for this repo.** `csrc/opus_gemm` already works this way, and #4622 makes exactly this change for the FlyDSL split-K HGEMM — same failure, same fix, with the reduce kernel and its tests already written. What is asked here is the ASM equivalent.

It costs more memory than a 4 KiB counter, but it removes the failure class rather than protecting it — including for hosts that never capture a graph.

The blocker is that the family ships precompiled as `hsa/gfx950/bf16gemm/*.co`, so it needs a rebuild rather than a source change.

## Scope

- `bf16gemm_*_splitk_*` in `hsa/gfx950/bf16gemm/`, reached through `aiter/ops/gemm_op_a16w16.py` → `csrc/py_itfs_cu/asm_gemm_a16w16.cu`.
- The FlyDSL split-K HGEMM has the same defect and is being fixed this way in #4622; those kernels are generated from the Python DSL, so they needed no rebuild. The ASM family is the remaining one.

## Reproduce

On `main`, MI355X (gfx950). **This wedges the selected GPU until the process is killed — do not run it on a shared runner.**

```python
# repro.py
import torch
import aiter.ops.gemm_op_a16w16 as G

KERNEL = "_ZN5aiter39bf16gemm_fp32bf16_tn_64x64_splitk_cleanE"
dev = torch.device("cuda:0")
x = torch.randn(64, 5120, dtype=torch.bfloat16, device=dev)
w = torch.randn(256, 5120, dtype=torch.bfloat16, device=dev)
out = torch.empty(64, 256, dtype=torch.float32, device=dev)

G.gemm_a16w16_asm(x, w, out, splitK=13, kernelName=KERNEL)
torch.cuda.synchronize()

stream = torch.cuda.Stream(device=dev)
with torch.cuda.stream(stream):
    # Warm the workspace for this stream and leave the counter at 2 -- the value
    # rocgdb found. Warming makes the capture below a cache hit, so no memset is
    # recorded, which is what happens in production.
    G.get_semaphore_workspace(dev).fill_(2)
stream.synchronize()

graph = torch.cuda.CUDAGraph()
with torch.cuda.graph(graph, stream=stream):
    G.gemm_a16w16_asm(x, w, out, splitK=13, kernelName=KERNEL)

print("replaying...", flush=True)
graph.replay()
torch.cuda.synchronize()          # never returns
print("returned -- the counter was restored")
```

```console
$ timeout 150 python3 repro.py; echo "exit=$?"
replaying...
exit=124
```

The seeded `2` is only a shortcut for getting the counter dirty deterministically. In production it gets there on its own, because the workspace is shared between launches that the kernel's own reset cannot separate once a graph replays them.

Without touching a GPU, the same defect is visible as "the graph contains no zero-fill" — capture, write a sentinel into the counter, replay, and read it back. That is `op_tests/test_gemm_a16w16_graph.py` in #4916, and it fails in about 3 s instead of hanging.
